packages/flutter_tools/lib/src/ios/code_signing.dart

```diff
@@ -116,229 +159,787 @@ Future<Map<String, String>?> getCodeSigningIdentityDevelopmentTeamBuildSetting({
     return null;
   }
 
-  if (_isNotEmpty(buildSettings['PROVISIONING_PROFILE'])) {
+  if (_isNotEmpty(buildSettings[_provisioningProfileBuildSettingName])) {
     return null;
   }
 
-  final String? developmentTeam = await _getCodeSigningIdentityDevelopmentTeam(
-    processManager: processManager,
-    platform: platform,
-    logger: logger,
+  final XcodeCodeSigningSettings settings = XcodeCodeSigningSettings(
     config: config,
+    logger: logger,
+    platform: platform,
+    processUtils: ProcessUtils(processManager: processManager, logger: logger),
+    fileSystem: fileSystem,
+    fileSystemUtils: fileSystemUtils,
     terminal: terminal,
-    shouldExitOnNoCerts: true,
+    plistParser: plistParser,
   );
 
-  if (developmentTeam == null) {
-    return null;
-  }
-
-  return <String, String>{_developmentTeamBuildSettingName: developmentTeam};
+  return settings._getCodeSigningBuildSettings();
 }
 
+/// Return the `DEVELOPMENT_TEAM` for automatic code-signing.
+/// This function should not be used for manual code-signing.
+///
+/// Find the `DEVELOPMENT_TEAM` from the saved `ios-signing-cert` or prompt the
```

Comment:
I'm assuming this comment is what the implementation does and not what the caller is expected to do? Consider:

```suggestion
/// This finds the `DEVELOPMENT_TEAM` from the saved `ios-signing-cert` or prompts the
```